sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -183,6 +187,60 @@ export class GlobalEndpointManager {
     }
   }
 
+  private refreshStaleUnavailableLocations(): void {
+    const now = Date.now();
+    let isReadRequestType: boolean = true;
+    this.updateLocationList(now, isReadRequestType);
+    this.cleanUnavailableLocationList(now, isReadRequestType);
+
+    isReadRequestType = false;
+    this.updateLocationList(now, isReadRequestType);
+    this.cleanUnavailableLocationList(now, isReadRequestType);
```

Comment:
This is just a note, not a change request, but something I often do when I want to annotate what a parameter is called instead of binding it to a variable is this:

```ts
this.updateLocationList(now, /* isReadRequest */ true);
```